File: src/problem12.hs
import Data.List

divisors :: Int -> [Int]
divisors n = (1 : filter ((==0) . rem n) [2..integerSqrt n]) ++ [n]

triangleNumber :: Int -> Int
triangleNumber n = n*(n+1) `div` 2

naturals :: [Int]
naturals = iterate (+1) 1

answer :: Maybe Int
answer = find (\x->(length (divisors x)*2) >= 500) $ map triangleNumber naturals

integerSqrt :: Int -> Int
integerSqrt = floor . sqrt . fromIntegral

File: src/problem12.py
import math

def divisors(n):
    i = 1
    divList = []
    for i in range(1,int(math.sqrt(n))):
        if n % i == 0:
            divList.append(i)
            divList.append(n / i)
    return divList

def problem12():
    i = 1
    while(len(divisors(i*(i+1)/2)) < 500):
        i += 1
    return (i*(i+1)/2)

print(problem12())
